profiling-procmacros/src/lib.rs

```diff
@@ -105,42 +105,19 @@ fn impl_block(
     }
 }
 
-#[cfg(feature = "profile-with-puffin")]
-fn impl_block(
-    body: &syn::Block,
-    _instrumented_function_name: &str,
-) -> syn::Block {
-    parse_quote! {
-        {
-            profiling::puffin::profile_function!();
-
-            #body
-        }
-    }
-}
-
-#[cfg(feature = "profile-with-optick")]
+#[cfg(any(
+    feature = "profile-with-puffin",
+    feature = "profile-with-optick",
+    feature = "profile-with-superluminal",
+    feature = "profile-with-tracy"
+))]
 fn impl_block(
     body: &syn::Block,
     _instrumented_function_name: &str,
 ) -> syn::Block {
     parse_quote! {
         {
-            profiling::optick::event!();
-
-            #body
-        }
-    }
-}
-
-#[cfg(feature = "profile-with-superluminal")]
-fn impl_block(
-    body: &syn::Block,
-    instrumented_function_name: &str,
-) -> syn::Block {
-    parse_quote! {
-        {
-            let _superluminal_guard = profiling::superluminal::SuperluminalGuard::new(#instrumented_function_name);
```

Comment:
I think the implementation here for superluminal is better than function_scope!'s implementation because it has a better way of ascertaining the function name. So I prefer we keep this implementation instead of just rerouting it to function_scope!()

I am not very strongly opinionated on if we consolidate optick, tracy, and puffin since they are identical implementations, however I lean towards just leaving these implementations alone. IMO the indirection is not worth the trivial duplication in code, especially since for two of the implementations we still need to keep implementations for them here.